client/api_client.js

```diff
@@ -10,9 +10,26 @@ var restClient = require('request-promise'),
  * @constructor
  */
 function ApiClient(urlBase, serviceId, apiKeyId) {
-  this.serviceId = serviceId;
-  this.apiKeyId = apiKeyId;
-  this.urlBase = urlBase;
+  if (arguments.length === 1 && !arguments[0].startsWith('http')) {
```

Comment:
i'd rather you wrote this to check for undefined values rather than arg length - feels like a more javascripthonic way to do it?